parsec/interfaces/ptg/ptg-compiler/jdf2c.c

```diff
@@ -3344,30 +3345,8 @@ static  void jdf_generate_deps_key_functions(const jdf_t *jdf, const jdf_functio
         
         for(vl = f->locals; vl != NULL; vl = vl->next) {
             if( local_is_parameter(f, vl) != NULL ) {
-                int have_min = 0;
-                if( vl->expr->op == JDF_RANGE ) {
-                    coutput("  int %s%s_min = %s;\n", JDF2C_NAMESPACE, vl->name, dump_expr((void**)vl->expr->jdf_ta1, &info));
-                    have_min = 1;
-                } else {
-                    if( vl->expr->local_variables != NULL ) {
-                        char *vname;
-                        if( asprintf(&vname, "%s%s_min", JDF2C_NAMESPACE, vl->name) <= 0 ) {
-                            fprintf(stderr, "Cannot allocate internal memory for the PTG compiler\n");
-                            exit(-1);
-                        }
-                        coutput("  int %s;\n", vname);
-                        jdf_generate_range_min_without_fn(jdf, vl->expr, vname, "(&"JDF2C_NAMESPACE"_assignments)");
```

Comment:
Is this function (`jdf_generate_range_min_without_fn`) still unecessary ?

Reply:
`jdf_generate_range_(min/max)_without_fn` are used by `jdf_generate_range_(min/max)`, which in turn are used by `jdf_generate_expression`. Those are useful for ranges in expressions that come from something else than a parameter, for which we don't have pre-computed min/max values...

They 'work' under the assumption that the leftmost value of the range is the minimum and the rightmost is the maximum, which is clearly an assumption that is not true anymore.

We can keep the old (unused, AFAICT) behavior, or remove them and set NULL as the minimum function and maximum function for range expression (so if we invoke dynamically, this will segfault/bus error), or we can replace them by real functions but then the only reliable way to find the minimum and maximum values are to iterate over the entire range and find the min/max... Which can be very costly.

As far as I can tell, we never dynamically invoke a range expression function.